nutanix/resource_nutanix_ndb_database_restore.go

```diff
@@ -0,0 +1,406 @@
+			return diag.FromErr(err)
+		}
+
+		if err := d.Set("tags", flattenDBTags(resp.Tags)); err != nil {
+			return diag.FromErr(err)
+		}
+
+		if err := d.Set("clone", resp.Clone); err != nil {
+			return diag.FromErr(err)
+		}
+
+		if err := d.Set("internal", resp.Internal); err != nil {
+			return diag.FromErr(err)
+		}
+
+		if err := d.Set("placeholder", resp.Placeholder); err != nil {
+			return diag.FromErr(err)
+		}
+
+		if err := d.Set("database_name", resp.Databasename); err != nil {
+			return diag.FromErr(err)
+		}
+
+		if err := d.Set("type", resp.Type); err != nil {
+			return diag.FromErr(err)
+		}
+
+		if err := d.Set("database_cluster_type", resp.Databaseclustertype); err != nil {
+			return diag.FromErr(err)
+		}
+
+		if err := d.Set("status", resp.Status); err != nil {
+			return diag.FromErr(err)
+		}
+
+		if err := d.Set("database_status", resp.Databasestatus); err != nil {
+			return diag.FromErr(err)
+		}
+
+		if err := d.Set("dbserver_logical_cluster_id", resp.Dbserverlogicalclusterid); err != nil {
+			return diag.FromErr(err)
+		}
+
+		if err := d.Set("time_machine_id", resp.Timemachineid); err != nil {
+			return diag.FromErr(err)
+		}
+
+		if err := d.Set("parent_time_machine_id", resp.Parenttimemachineid); err != nil {
+			return diag.FromErr(err)
+		}
+
+		if err := d.Set("time_zone", resp.Timezone); err != nil {
+			return diag.FromErr(err)
+		}
+
+		if err := d.Set("info", flattenDBInfo(resp.Info)); err != nil {
+			return diag.FromErr(err)
+		}
+
+		if err := d.Set("group_info", resp.GroupInfo); err != nil {
+			return diag.FromErr(err)
+		}
+
+		if err := d.Set("metadata", flattenDBInstanceMetadata(resp.Metadata)); err != nil {
+			return diag.FromErr(err)
+		}
+
+		if err := d.Set("metric", resp.Metric); err != nil {
+			return diag.FromErr(err)
+		}
+
+		if err := d.Set("category", resp.Category); err != nil {
+			return diag.FromErr(err)
+		}
+
+		if err := d.Set("parent_database_id", resp.ParentDatabaseID); err != nil {
+			return diag.FromErr(err)
+		}
+
+		if err := d.Set("parent_source_database_id", resp.ParentSourceDatabaseID); err != nil {
+			return diag.FromErr(err)
+		}
+
+		if err := d.Set("lcm_config", flattenDBLcmConfig(resp.Lcmconfig)); err != nil {
+			return diag.FromErr(err)
+		}
+
+		if err := d.Set("time_machine", flattenDBTimeMachine(resp.TimeMachine)); err != nil {
+			return diag.FromErr(err)
+		}
+
+		if err := d.Set("dbserver_logical_cluster", resp.Dbserverlogicalcluster); err != nil {
+			return diag.FromErr(err)
+		}
+
+		if err := d.Set("database_nodes", flattenDBNodes(resp.Databasenodes)); err != nil {
+			return diag.FromErr(err)
+		}
+
+		if err := d.Set("linked_databases", flattenDBLinkedDbs(resp.Linkeddatabases)); err != nil {
+			return diag.FromErr(err)
+		}
+	}
+
+	return nil
+}
+
+func resourceNutanixNDBDatabaseRestoreUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	return nil
```

Comment:
Same problem here we need to have logic similar to create... because if user wants to restore at a particular point after making one restore he doesn't need to write another block.

Reply:
implemented restore_count to restore agian at same point of time or with same snapshot id.